### PR TITLE
Add maxAnisotropy property

### LINKED ISSUE #696: Anisotropic clamping for SamplerDescriptor
We have the anisotropic extension string, but no associated API to set the clamp.
I wonder how this will be specified: just adding a member to `SamplerDescriptor` may look like this is a part of the core spec.

### PR DESCRIPTION
**Pull request by kvark**
Closes #696

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -1786,6 +1786,7 @@ dictionary GPUSamplerDescriptor : GPUObjectDescriptorBase {
     float lodMinClamp = 0;
     float lodMaxClamp = 0xffffffff; // TODO: What should this be? Was Number.MAX_VALUE.
     GPUCompareFunction compare;
+    unsigned short maxAnisotropy = 1;
```
[the remark this answers is not in the document]
Comment (kvark) [reply spec/index.bs:1789]: yeah, I don't think there is a strong reason to have it as float. Integer should be fine

File: spec/index.bs [hunk 2]
```diff
@@ -1801,8 +1802,10 @@ dictionary GPUSamplerDescriptor : GPUObjectDescriptorBase {
     maximum levels of detail, respectively, used internally when sampling a texture.
 - If {{GPUSamplerDescriptor/compare}} is provided, the sampler will be a comparison sampler with the specified
     {{GPUCompareFunction}}.
+- {{GPUSamplerDescriptor/maxAnisotropy}} specifies the maximum anisotropy value clamp used by the sampler.
```
[the remark this answers is not in the document]
Comment (kvark) [reply spec/index.bs:1805]: I believe the limits anisotropy in Vulkan is 16 for all devices that support it. So the way it's specced right now in the PR is correct, there is no additional clamping.

[the remark this answers is not in the document]
Comment (kainino0x) [reply spec/index.bs:1805]: But wouldn't we additionally clamp it to 1, if anisotropic filtering isn't supported?

[the remark this answers is not in the document]
Comment (kvark) [reply spec/index.bs:1805]: I don't think we have to. As mentioned on the call, the pattern is implementation dependent. There is no need to specify this clamping to 1 on some small number of platforms if they are allowed to have the pattern all being the same texel anyway :)

[the remark this answers is not in the document]
Comment (kainino0x) [reply spec/index.bs:1805]: Right, gotcha.

File: spec/index.bs [hunk 3]
```diff
@@ -1918,6 +1921,7 @@ enum GPUCompareFunction {
         - |descriptor|.{{GPUSamplerDescriptor/lodMinClamp}} is greater than or equal to 0.
         - |descriptor|.{{GPUSamplerDescriptor/lodMaxClamp}} is greater than or equal to
             |descriptor|.{{GPUSamplerDescriptor/lodMinClamp}}.
+        - |descriptor|.{{GPUSamplerDescriptor/maxAnisotropy}} is between 1 and 16, inclusive.
```
Comment (jdashg) [inline spec/index.bs:1924]: I don't think this needs a specified max, even though it's not larger than 16x in practice.

File: spec/index.bs [hunk 4]
```diff
@@ -1803,6 +1803,7 @@ dictionary GPUSamplerDescriptor : GPUObjectDescriptorBase {
 - If {{GPUSamplerDescriptor/compare}} is provided, the sampler will be a comparison sampler with the specified
     {{GPUCompareFunction}}.
 - {{GPUSamplerDescriptor/maxAnisotropy}} specifies the maximum anisotropy value clamp used by the sampler.
+    Most implementations support values in range between 1 and 16, inclusive.
```
Comment (kainino0x) [inline spec/index.bs:1806]: Should be `Note:`?

Comment (kvark) [reply spec/index.bs]: I don't know how to put a note into this block. Moved outside now.

File: spec/index.bs [hunk 5]
```diff
@@ -1801,8 +1802,12 @@ dictionary GPUSamplerDescriptor : GPUObjectDescriptorBase {
     maximum levels of detail, respectively, used internally when sampling a texture.
 - If {{GPUSamplerDescriptor/compare}} is provided, the sampler will be a comparison sampler with the specified
     {{GPUCompareFunction}}.
+- {{GPUSamplerDescriptor/maxAnisotropy}} specifies the maximum anisotropy value clamp used by the sampler.
+
+Note: most implementations support {{GPUSamplerDescriptor/maxAnisotropy}} values in range between 1 and 16, inclusive.
```
Comment (kainino0x) [inline spec/index.bs:1807]: I think it should work if it's indented enough (this list is only indented by 1 but it's probably treated as 4?, so the next line might need 6 or 8)

Or you can fix the list so it's indented by 4 and the continuations are indented by 8.
```suggestion
        Note: most implementations support {{GPUSamplerDescriptor/maxAnisotropy}} values in range between 1 and 16, inclusive.
```

Comment (kainino0x) [reply spec/index.bs:1807]: Actually this list is not indented so it can probably go down to 0 and have the Note be at 4.

**Pull request merged**